src/pipeline_factory.cpp

```diff
@@ -111,4 +111,12 @@ void PipelineFactory::revalidatePipelines(ModelManager& manager) {
         }
     }
 }
+
+void PipelineFactory::getPipelinesStatuses(std::map<std::string, PipelineDefinitionStatus>& pipelineStatuses) {
+    std::shared_lock lock(definitionsMtx);
+    for (auto const& [pipelineName, definition] : definitions) {
+        auto status = std::pair<std::string, PipelineDefinitionStatus>(pipelineName, definition->getStatus());
+        pipelineStatuses.insert(status);
```

Comment:
you could either:
std::map.insert(std::move(status))
or
std::map.emplace(std::pair<std::string, PipelineDefinitionStatus>(pipelineName, definition->getStatus()))

Additionally if you really don't care about the order of the map then use std::unordered_map instead.